### ISSUE TITLE
[ENH] Refactor load_from_tsfile_to_dataframe to use functionality for .tsf files too

### ISSUE THREAD

**Issue by ltsaprounis** [alan-turing-institute/sktime#1881]
[Monash Time Series Forecasting Repository](https://forecastingdata.org) has introduced a new file format for storing time series data for forecasting problems called .tsf, inspired by sktime's .ts format. The repository has a lot of nice/popular benchmark datasets, including Australian tourism datasets and 3/5 of the M competitions.

I suggest we refactor [load_from_tsfile_to_dataframe](https://github.com/alan-turing-institute/sktime/blob/79cc513346b1257a6f3fa8e4ed855b5a2a7de716/sktime/utils/data_io.py#L33) slightly to use the overlapping features in a new function called `load_from_tsffile_to_dataframe` (we can work on the name). 

A .tsf reader is available from the Monash repository team here: 
https://github.com/rakshitha123/TSForecasting/blob/master/utils/data_loader.py

@fkiraly @TonyBagnall @rakshitha123

**Comment by ltsaprounis** [comment]
Draft PR coming soon!

**Comment by fkiraly** [comment]
Yes, some investigation on which common functionality can be factored out, or on design would be useful in prep for the meeting!
